File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/service/impl/DefaultHashService.java
package com.shterneregen.securelan.crypto.service.impl;

import com.shterneregen.securelan.crypto.service.HashService;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.HexFormat;
import java.util.Objects;

public final class DefaultHashService implements HashService {
    @Override
    public byte[] sha256(byte[] input) {
        Objects.requireNonNull(input, "input");
        try {
            return MessageDigest.getInstance("SHA-256").digest(input);
        } catch (NoSuchAlgorithmException exception) {
            throw new IllegalStateException("SHA-256 is not available", exception);
        }
    }

    @Override
    public String sha256Hex(byte[] input) {
        return HexFormat.of().formatHex(sha256(input));
    }
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/service/impl/DefaultHashService.kt
package com.shterneregen.securelan.crypto.service.impl

import com.shterneregen.securelan.crypto.service.HashService
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import java.util.HexFormat
import java.util.Objects

class DefaultHashService : HashService {
    override fun sha256(input: ByteArray): ByteArray {
        Objects.requireNonNull(input, "input")
        try {
            return MessageDigest.getInstance("SHA-256").digest(input)
        } catch (exception: NoSuchAlgorithmException) {
            throw IllegalStateException("SHA-256 is not available", exception)
        }
    }

    override fun sha256Hex(input: ByteArray): String = HexFormat.of().formatHex(sha256(input))
}
